### PR TITLE
:electron: Notarize the mac desktop app

### PR DESCRIPTION
**Pull request by MikesGlitch**
Should close these: 
https://github.com/actualbudget/actual/issues/3259
https://github.com/actualbudget/actual/issues/3088

Tested with this artifact: https://github.com/MikesGlitch/actual/actions/runs/10496563766

### CODE REVIEW

File: packages/desktop-electron/package.json [hunk 1]
```diff
@@ -31,9 +31,6 @@
       "hardenedRuntime": true,
       "gatekeeperAssess": false,
       "artifactName": "${productName}-mac-${arch}.${ext}",
-      "notarize": {
```
Comment (MikesGlitch) [inline packages/desktop-electron/package.json:34]: Docs say to remove this from config and into env variables

File: .github/workflows/electron-pr.yml [hunk 2]
```diff
@@ -43,7 +43,17 @@ jobs:
           sudo flatpak install org.electronjs.Electron2.BaseApp/x86_64/23.08 -y
       - name: Set up environment
         uses: ./.github/actions/setup
+      - name: Build Electron for Mac
+        if: ${{ startsWith(matrix.os, 'macos') }}
+        run: ./bin/package-electron
+        env:
+          CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD }}
```
Comment (youngcw) [inline .github/workflows/electron-pr.yml:50]: iirc we didn't add the notarization here because it caused issues with forks.  Maybe @MatissJanis remembers

Comment (MikesGlitch) [reply .github/workflows/electron-pr.yml:50]: This is for testing - no reviews yet!!

File: .github/workflows/electron-master.yml [hunk 3]
```diff
@@ -48,13 +48,17 @@ jobs:
...
         env:
-          # CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD }}
-          # CSC_LINK: ${{ secrets.CSC_LINK }}
+          CSC_KEY_PASSWORD: ${{ secrets.CSC_KEY_PASSWORD }}
+          CSC_LINK: ${{ secrets.CSC_LINK }}
           APPLE_ID: ${{ secrets.APPLE_ID }}
           APPLE_APP_SPECIFIC_PASSWORD: ${{ secrets.APPLE_APP_SPECIFIC_PASSWORD }}
+      - name: Build Electron
+        if: ${{ ! startsWith(matrix.os, 'macos') }}
+        run: ./bin/package-electron
```
Comment (MikesGlitch) [inline .github/workflows/electron-master.yml:61]: When Mac, build for mac with the mac env variables. Otherwise build without env variables.

**Review by MatissJanis — APPROVED** [review]

**Comment by yoyotogblo** [comment]
Thanks for this @MikesGlitch . Are you also able to add it to the electron-pr.yml? Would be great to have the PR electron files signed also.

**Comment by MikesGlitch** [comment]
> Thanks for this @MikesGlitch . Are you also able to add it to the electron-pr.yml? Would be great to have the PR electron files signed also.

It's not possible - if someone PR's against this repo with malicious intent, they could take our developer signing and pretend to be Actual. Code signing can only happen if we've approved the code.

**Pull request merged**